modules/apps/27-interchain-accounts/controller/types/msgs.go

```diff
@@ -75,14 +73,14 @@ func (msg MsgSendTx) ValidateBasic() error {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "failed to parse owner address: %s", msg.Owner)
 	}
 
-	if msg.TimeoutHeight.IsZero() && msg.TimeoutTimestamp == 0 {
-		return sdkerrors.Wrap(channelerrors.ErrInvalidTimeout, "msg timeout height and msg timeout timestamp cannot both be 0")
-	}
-
 	if err := msg.PacketData.ValidateBasic(); err != nil {
 		return sdkerrors.Wrap(err, "invalid interchain account packet data")
 	}
 
+	if msg.RelativeTimeout == 0 {
+		return errors.New("relative timeout cannot be zero")
```

Comment:
We have to use sdkerrors with a registered error, otherwise the error returned to the user will be wiped and replaced with "internal error". See [godoc](https://github.com/cosmos/cosmos-sdk/blob/main/errors/abci.go#L25)

Reply:
thanks! I was uncertain about how this behaved, thanks for the link!

Reply:
Looks like this still needs to be updated